src/FurCoNZ.Web/Controllers/AccountController.cs

```diff
@@ -33,18 +36,30 @@ public async Task<IActionResult> Index(CancellationToken cancellationToken = def
             return View(new AccountViewModel(await _userService.GetCurrentUserAsync(cancellationToken)));
         }
 
-        [HttpPost]
+        [HttpGet, HttpPost]
         public async Task<IActionResult> UpdateAccount(AccountViewModel account, CancellationToken cancellationToken = default)
         {
             var user = await _userService.GetCurrentUserAsync(cancellationToken);
+            var hasRedirect = HttpContext.Session.Keys.Contains(RedirectAfterDetailsSessionKey);
+            if (ModelState.IsValid) {
+
+                user.Name = account.Name;
+                // TODO: Require email validation of this change
```

Comment:
This can be the login server's responsibility most of the time if we like. The login server will be updated to pass a new claim confirming the email address has been verified.

If the user opts not to share their email address, or the email address from the IdP is marked as not verified, then we will still need to validate it here, yes.